bids/analysis/transformations/compute.py

```diff
@@ -49,11 +55,18 @@ def _transform(self, var, model='spm', derivative=False, dispersion=False,
         elif model != 'fir':
             raise ValueError("Model must be one of 'spm', 'glover', or 'fir'.")
 
-        convolved = hrf.compute_regressor(vals, model, onsets,
-                                          fir_delays=fir_delays, min_onset=0)
+        oversampling = np.ceil(
+            1 / min([np.ediff1d(var.onset).min(), var.duration.min()])
+            * (1/sampling_rate))
```

Comment:
Is `var.onset` guaranteed to be sorted? If not, `np.ediff1d([0.2, 0.6, 0.1])  == array([ 0.4, -0.5])`, meaning you get `-0.5` instead of `0.1`, which I suspect you'd prefer.

Also, I think this will be a little easier to parse if split into two steps:

```Python
min_interval = min(np.ediff1d(np.sort(var.onset)).min(),
                   var.duration.min())
oversampling = np.ceil(1 / (min_interval * sampling_rate))
```

Reply:
I think it should be, although I'm not sure its guaranteed. Could always sort first.